module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,181 @@
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+
+                let pollSubmissionDocument = await this.pollSubmissionDocuments
+                (
+                   {
+                       userId: userId,
+                       pollId: pollId
+                   }
+                )
+
+                if (pollSubmissionDocument.length > 0) {
+                    throw new Error(messageConstants.apiResponses.MULTIPLE_SUBMISSIONS_NOT_ALLOWED)
+                }
+
+                let pollDocument = await pollsHelper.pollDocuments
+                (
+                    {
+                        _id: pollId
+                    },
+                    [
+                        "name",
+                        "numberOfResponses",
+                        "result"
+                    ]
+                )
+                 
+                 pollSubmissionDocument = {
+                    pollName: pollDocument[0].name,
+                    pollId: pollId,
+                    submittedAt: new Date(),
+                    responses: responseObject,
+                    isDeleted: false,
+                    userId: userId,
+                    status: "active"
+                }
+
+                await database.models.pollSubmissions.create(pollSubmissionDocument);
+
+                let result = pollDocument[0]["result"] ? pollDocument[0]["result"] : {}
+                
+                Object.values(responseObject).forEach(singleResponse => {
+                    if (!result[singleResponse.qid]) {
+                        result[singleResponse.qid] = {
+                            question: singleResponse.question,
+                            responses: {}
+                        }
+                    }
+                
+                    if (Array.isArray(singleResponse.label)) {
+                        singleResponse.label.forEach(label => {
+                            if (!result[singleResponse.qid].responses[label]) {
+                                result[singleResponse.qid].responses[label] = 1;
+                            }
+                            else {
+                                result[singleResponse.qid].responses[label] = result[singleResponse.qid].responses[label] + 1;
+                            }
+                        })
+                    }
+                    else {
+                        if (!result[singleResponse.qid].responses[singleResponse.label]) {
+                            result[singleResponse.qid].responses[singleResponse.label] = 1;
+                        }
+                        else {
+                            result[singleResponse.qid].responses[singleResponse.label] = result[singleResponse.qid].responses[singleResponse.label] + 1;
+                        }
+                    }
+                })
+
+                await database.models.polls.updateOne
```

Comment:
@deepakr123 - Use helper function to update poll results.

Reply:
resolved